packages/sdk/src/fetcher/search/index.ts

```diff
@@ -0,0 +1,40 @@
+import { KPIToken } from "../../entities/kpi-token";
+import { KPITokensObject, transformInKPITokensObject } from "../serializer";
+
+export function searchKPItokens(
+    searchQuery: string,
+    allKPITokens: KPITokensObject
+) {
+    const filteredTokens = Object.values(allKPITokens).filter((token) =>
+        tokenSpecificationIncludesQuery(token, searchQuery)
+    );
+
+    return transformInKPITokensObject(filteredTokens);
+}
+
+function tokenSpecificationIncludesQuery(token: KPIToken, searchQuery: string) {
+    if (!token.hasOwnProperty("specification")) return;
```

Comment:
Why is this a check?

Reply:
I think we can remove the check. Thanks for pointing out.